### PR TITLE
[RIP-9] Add the introduction of the Operations_Producer in RocketMQ

### PR DESCRIPTION
**Pull request by frayds**
## What is the purpose of the change

Add description section for the Operations_Producer to RocketMQ Develop Guide:

## Brief changelog

* Add description section for Operations_Producer to English RocketMQ Develop Guide

## Verifying this change

Browse the section 1 in the best_practice.md

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: docs/en/Operations_Producer.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+### Producer
+----
+##### 1 Message Sending Tips
+###### 1.1 The Use of Tags
+One instance should use one topic as much as possible and the subtype of messages can be marked by tags. Tag provides extra flexibility to users. In the consume subscribing process, the messages filtering can only be handled by using tags when the tags are specified in the message sending process: `message.setTags("TagA")`
+###### 1.2 The Use of Keys
+A business  key can be set in one message and it will be easier to look up the message on a broker server to diagnose issues during development. Each message will be created index(hash index) by server, instance can query the content of this message by topic and key and who consumes the message.Because of the hash index, make sure that the key should be unique in order to avoid potential hash index conflict.
```
Comment (wlliqipeng) [inline docs/en/Operations_Producer.md]: The first sentence seems to be inaccurate.

File: docs/en/Operations_Producer.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+message.setKeys(orderId);
+```
+###### 1.3 The Log Print
+When sending a message,no matter success or fail, a message log must be printed which contains SendResult and Key. It is assumed that we will always get SEND_OK if no exception is thrown. Below is a list of descriptions about each status:
+* SEND_OK
+
+SEND_OK means sending message successfully. SEND_OK does not mean it is reliable. To make sure no message would be lost, you should also enable SYNC_MASTER or SYNC_FLUSH.
+* FLUSH_DISK_TIMEOUT
+
+FLUSH_DISK_TIMEOUT means sending message successfully but the Broker flushing the disk with timeout. In this kind of condition, the Broker has saved this message, this message will be lost only if the Broker was down. The FlushDiskType and SyncFlushTimeout could be specified in MessageStoreConfig. If the Broker set MessageStoreConfig’s FlushDiskType=SYNC_FLUSH(default is ASYNC_FLUSH), and the Broker doesn’t finish flushing the disk within MessageStoreConfig’s syncFlushTimeout(default is 5 secs), you will get this status.
```
Comment (wlliqipeng) [inline docs/en/Operations_Producer.md]: the Broker has saved this message **in memory**

File: docs/en/Operations_Producer.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* FLUSH_SLAVE_TIMEOUT
+
+FLUSH_SLAVE_TIMEOUT means sending messages successfully but the slave Broker does not finish synchronizing with the master. In this kind of condition, the Broker has saved this message, this message will be lost only if the Broker was down. If the Broker’s role is SYNC_MASTER(default is ASYNC_MASTER), and the slave Broker doesn’t finish synchronizing with the master within the MessageStoreConfig’s syncFlushTimeout(default is 5 secs), you will get this status.
```
Comment (wlliqipeng) [inline docs/en/Operations_Producer.md]: ‘In this kind of condition, the Broker has saved this message, this message will be lost only if the Broker was down’ this sentence is recommended to delete

File: docs/en/Operations_Producer.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+* SLAVE_NOT_AVAILABLE
+
+SLAVE_NOT_AVAILABLE means sending messages successfully but no slave Broker configured. In this kind of condition, the Broker has saved this message, this message will be lost only if the master Broker was down. If the Broker’s role is SYNC_MASTER(default is ASYNC_MASTER), but no slave Broker is configured, you will get this status.
```
Comment (wlliqipeng) [inline docs/en/Operations_Producer.md]: ‘In this kind of condition, the Broker has saved this message, this message will be lost only if the master Broker was down.’ the sentence is recommended to delete

File: docs/en/Operations_Producer.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+The reasons why the retry strategy using database have not realized by the RocketMQ client will be explained below: Firstly, the design mode of the RocketMQ client is stateless mode. It means that the client is designed to be horizontally scalable at each level and the consumption of the client to physical resources is only CPU, memory and web. Then, if a key-value memory module is realized by the client itself, the Asyn-Saving strategy will be utilized in consideration of the high resource consumption of the Syn-Saving strategy. However, given that operations staff does not manage the client shutoff, some special commands, such as kill -9, may be used which will lead to the lost of message because of no saving in time. Furthermore, the physical resource running Producer is not appropriate to save some significant data because of low reliability. Above all, the retry process should be controlled by program itself.
```
Comment (wlliqipeng) [inline docs/en/Operations_Producer.md]: It is seems better to replace web with network

**Comment by ShannonDing** [comment]
@frayds Could you please check these suggestions again?

**Pull request merged**